### ISSUE TITLE
Bad linux binaries for v1.0.6

### ISSUE THREAD

**Issue by a-s-o** [openshift/origin#5347]
Hello,

The release page says that the linux binaries were not tested and it looks like they may be broken. Here is what I get after extraction:

![output](http://imgur.com/xoR3yPc.png)

As expected, only kubelet and oc work from the command line. 

I am on Ubuntu 15.04 64-bit. Origin v 1.0.5 archive works fine on my system

**Comment by Miciah** [comment]
What tool did you use to extract the files? Most likely it does not support symlinks.  It works for me using `tar` from tar-1.26-29.el7.x86_64 on Red Hat Enterprise Linux 7:
```console
% mkdir -p /tmp/openshift-origin-v1.0.6-36-g0d87e4e-0d87e4e-linux-386
% cd /tmp/openshift-origin-v1.0.6-36-g0d87e4e-0d87e4e-linux-386
% tar xf ~/Downloads/openshift-origin-v1.0.6-36-g0d87e4e-0d87e4e-linux-386.tar.gz
% ls -s
total 565M
72M kube-apiserver  72M kube-controller-manager  32M kubectl  72M kubelet  72M kube-proxy  72M kube-scheduler  72M oadm  32M oc  72M openshift
% tar tvf ~/Downloads/openshift-origin-v1.0.6-36-g0d87e4e-0d87e4e-linux-386.tar.gz
drwx------ root/root         0 2015-09-15 17:29 ./
-rwxr-xr-x root/root  75135520 2015-09-15 17:29 ./kubelet
-rwxr-xr-x root/root  32904576 2015-09-15 17:29 ./oc
hrwxr-xr-x root/root         0 2015-09-15 17:29 ./kubectl link to ./oc
hrwxr-xr-x root/root         0 2015-09-15 17:29 ./kube-scheduler link to ./kubelet
hrwxr-xr-x root/root         0 2015-09-15 17:29 ./kube-proxy link to ./kubelet
hrwxr-xr-x root/root         0 2015-09-15 17:29 ./openshift link to ./kubelet
hrwxr-xr-x root/root         0 2015-09-15 17:29 ./kube-controller-manager link to ./kubelet
hrwxr-xr-x root/root         0 2015-09-15 17:29 ./oadm link to ./kubelet
hrwxr-xr-x root/root         0 2015-09-15 17:29 ./kube-apiserver link to ./kubelet
```
The 64-bit Linux binaries are packaged the same way (with every executable but `kubelet` and `oc` being a symlink to one of those two) in the 1.0.6 release:
```console
% tar tvf ~/Downloads/openshift-origin-v1.0.6-2695cdc-linux-amd64.tar.gz
drwx------ root/root         0 2015-09-13 16:35 ./
-rwxr-xr-x root/root  91424233 2015-09-13 16:34 ./kubelet
-rwxr-xr-x root/root  39931294 2015-09-13 16:34 ./oc
^Phrwxr-xr-x root/root         0 2015-09-13 16:34 ./kubectl link to ./oc
hrwxr-xr-x root/root         0 2015-09-13 16:34 ./kube-scheduler link to ./kubelet
hrwxr-xr-x root/root         0 2015-09-13 16:34 ./kube-proxy link to ./kubelet
hrwxr-xr-x root/root         0 2015-09-13 16:34 ./openshift link to ./kubelet
hrwxr-xr-x root/root         0 2015-09-13 16:34 ./kube-controller-manager link to ./kubelet
hrwxr-xr-x root/root         0 2015-09-13 16:34 ./oadm link to ./kubelet
hrwxr-xr-x root/root         0 2015-09-13 16:34 ./kube-apiserver link to ./kubelet
```
This is a change from the 1.0.5 release:
```console
% tar tvf ~/Downloads/openshift-origin-v1.0.5-96963b6-linux-amd64.tar.gz
drwx------ root/root         0 2015-08-24 18:15 ./
-rwxr-xr-x root/root  88171576 2015-08-24 18:15 ./oadm
-rwxr-xr-x root/root  88171576 2015-08-24 18:14 ./openshift
-rwxr-xr-x root/root  39862537 2015-08-24 18:14 ./oc
```

**Comment by a-s-o** [comment]
Oh, I see. I used a node.js [script](github.com/kevva/download) which uses [node-tar](github.com/isaacs/node-tar) internally and I also tried extracting with the ubuntu archive manager. You are right it is an issue with those tools; using `tar` from the command line worked fine. 

So, the new binaries are overloaded based on which command name is used to call them but internally they are just those two things. 

Thanks for the explanation. I will update my scripts.
